From 28712799d934459b0d8a119e6a7a629def4016e1 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 9 Nov 2021 09:44:55 +0100
Subject: [PATCH] BUG: fix gcc version detection fallback for Mingw32CCompiler

In case gcc_version is None it would try to parse the output, but
this code was never ported to Python 3, so it matches str with bytes
which raises a TypeError.

Fix by decoding the gcc output via the text option (available since py3.7)

Also fixes the reference to CalledProcessError
---
 numpy/distutils/mingw32ccompiler.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/numpy/distutils/mingw32ccompiler.py b/numpy/distutils/mingw32ccompiler.py
index 82d296434..ac745fbaf 100644
--- a/numpy/distutils/mingw32ccompiler.py
+++ b/numpy/distutils/mingw32ccompiler.py
@@ -66,8 +66,10 @@ def __init__ (self,
         # get_versions methods regex
         if self.gcc_version is None:
             try:
-                out_string  = subprocess.check_output(['gcc', '-dumpversion'])
-            except (OSError, CalledProcessError):
+                out_string = subprocess.check_output(
+                    ['gcc', '-dumpversion'],
+                    text=True)
+            except (OSError, subprocess.CalledProcessError):
                 out_string = ""  # ignore failures to match old behavior
             result = re.search(r'(\d+\.\d+)', out_string)
             if result:
-- 
2.33.0

